src/iblock.cpp

```diff
@@ -10,51 +10,53 @@
 #include <boost/algorithm/string.hpp>
 #include <util/ixmlfile.h>
 #include <util/logstream.h>
+#include <stdio.h>
 #include "iblock.h"
 #include "md4block.h"
 #include "tx3block.h"
 
+#if (!_MSC_VER)
+    #include <cerrno>
+#endif
+
 using namespace std::chrono_literals;
 using namespace util::log;
 
 namespace mdf::detail {
 
 int64_t GetFilePosition(std::FILE *file) {
-  std::fpos_t curr = {};
-  auto get = std::fgetpos(file, &curr);
-  if (get != 0) {
+#if (_MSC_VER)
+  auto pos _ftelli64(file);
```

Comment:
yes I forgott it here. I am currently not able to test on Windows so it might happen. I will rebase this branch on your one once all boost dependencies are out. Then I try to get it compiled on linux